### PR TITLE
Java configurations have been added to the project. 

### PR DESCRIPTION
**Pull request by LicenserX**
Package controller has been rewritten. Annotations have been added in some classes.

### CODE REVIEW

File: src/main/java/com/epam/lab/paymentsystem/controller/UserController.java [hunk 1]
```diff
@@ -3,41 +3,39 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
 
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import java.io.IOException;
-
+@Controller
```
Comment (nikandtos1) [inline src/main/java/com/epam/lab/paymentsystem/controller/UserController.java]: If you use java config, you dont need Controller, Service, Repository annotation.

File: src/main/java/com/epam/lab/paymentsystem/controller/IndexController.java [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
 
+@Controller
+public class IndexController{
     private static final String INDEX_PAGE = "index";
 
-    @Override
-    protected ModelAndView handleRequestInternal(HttpServletRequest req, HttpServletResponse resp) throws Exception {
-        ModelAndView modelAndView = new ModelAndView(INDEX_PAGE);
-        return modelAndView;
+    @RequestMapping(value = "/", method = RequestMethod.GET)
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/controller/IndexController.java]: I would like you to use @GetMapping, @PostMapping and etc.

File: src/main/java/com/epam/lab/paymentsystem/controller/UserController.java [hunk 3]
```diff
@@ -3,41 +3,39 @@
...
+        user.setName(userName);
+        user.setLogin(userLogin);
+        user.setPassword(userPassword);
 
         try {
             userService.addUser(user);
-            resp.sendRedirect(req.getContextPath() + "/");
         } catch (LoginAlreadyExistsException e) {
-            req.setAttribute("submit", e.getMessage());
+            return "redirect:/";
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/controller/UserController.java]: Please, use constants

File: src/main/java/com/epam/lab/paymentsystem/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -7,10 +7,15 @@
...
 import com.epam.lab.paymentsystem.exception.LoginAlreadyExistsException;
 import com.epam.lab.paymentsystem.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
+@Service
 public class UserServiceImpl implements UserService {
 
+    @Autowired
     private final UserDAO userDAO;
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/service/impl/UserServiceImpl.java]: `final` is redundant here, also sometimes autowiring might not work due to `final` (like in the case of autowiring through setter)

File: src/main/java/com/epam/lab/paymentsystem/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -7,10 +7,15 @@
...
+    @Autowired
     private final RoleDAO roleDAO;
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/service/impl/UserServiceImpl.java]: Same as above

File: src/main/java/com/epam/lab/paymentsystem/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -7,15 +7,12 @@
 import com.epam.lab.paymentsystem.entities.enums.Roles;
 import com.epam.lab.paymentsystem.exception.LoginAlreadyExistsException;
 import com.epam.lab.paymentsystem.service.UserService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class UserServiceImpl implements UserService {
 
-    @Autowired
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/service/impl/UserServiceImpl.java]: Why did you remove @Autowired?

Comment (LicenserX) [reply src/main/java/com/epam/lab/paymentsystem/service/impl/UserServiceImpl.java]: UserDAO, RoleDAO are use in constratort UserServiceImpl(UserDAO userDAO, UserDAO userDAO) that's why is not necessary to mark it @Autowired.

File: src/main/java/com/epam/lab/paymentsystem/controller/UserController.java [hunk 7]
```diff
@@ -3,41 +3,38 @@
...
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import java.io.IOException;
-
+@Controller
 public class UserController {
-
     private static final String REGISTRATION_PAGE = "registration";
+    private static final String ROOT = "redirect:/";
```
Comment (nikandtos) [inline src/main/java/com/epam/lab/paymentsystem/controller/UserController.java]: redirect isn't a part of mapping.  Separate it please.

File: src/main/java/com/epam/lab/paymentsystem/service/impl/UserDetailsServiceImpl.java [hunk 8]
```diff
@@ -6,7 +6,9 @@
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
 
+@Service
 public class UserDetailsServiceImpl implements UserDetailsService {
 
     private final UserDAO userDAO;
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/service/impl/UserDetailsServiceImpl.java]: Redundant `final`
